src/mockchain/lib/Convex/MockChain.hs

```diff
@@ -475,7 +475,7 @@ instance Monad m => MonadMockchain (MockchainT m) where
     pure a
 
 instance Monad m => MonadUtxoQuery (MockchainT m) where
-  utxosByPaymentCredentials cred = onlyCredentials cred <$> utxoSet
+  utxosByPaymentCredentials cred = onlyCredentials cred . fmap (const Nothing) <$> utxoSet
```

Comment:
We should use `queryDatumFromHash` (from `MonadDatumQuery`) to resolve the datum hash